pkg/resources/digitalocean/resources.go

```diff
@@ -121,6 +148,110 @@ func listVolumes(cloud fi.Cloud, clusterName string) ([]*resources.Resource, err
 	return resourceTrackers, nil
 }
 
+func getAllVolumesByRegion(cloud *Cloud, region string) ([]godo.Volume, error) {
+	allVolumes := []godo.Volume{}
+
+	opt := &godo.ListOptions{}
+	for {
+		volumes, resp, err := cloud.Volumes().ListVolumes(context.TODO(), &godo.ListVolumeParams{
+			Region:      region,
+			ListOptions: opt,
+		})
+
+		if err != nil {
+			return nil, err
+		}
+
+		allVolumes = append(allVolumes, volumes...)
+
+		if resp.Links == nil || resp.Links.IsLastPage() {
+			break
+		}
+
+		page, err := resp.Links.CurrentPage()
+		if err != nil {
+			return nil, err
+		}
+
+		opt.Page = page + 1
+	}
+
+	return allVolumes, nil
+}
+
+func listDNS(cloud fi.Cloud, clusterName string) ([]*resources.Resource, error) {
+	c := cloud.(*Cloud)
+
+	domains, _, err := c.Client.Domains.List(context.TODO(), &godo.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list domains: %s", err)
+	}
+
+	var domainName string
+	for _, domain := range domains {
+		if strings.HasSuffix(clusterName, domain.Name) {
+			domainName = domain.Name
```

Comment:
Might be a good idea to check here that it isn't ambiguous - i.e. that `domainName == ""`